src/snowflake/snowpark/table.py

```diff
@@ -682,8 +682,8 @@ def drop_table(self, if_exists: bool = False) -> None:
         Note that subsequent operations such as :meth:`DataFrame.select`, :meth:`DataFrame.collect` on this ``Table`` instance and the derived DataFrame will raise errors because the underlying
         table in the Snowflake database no longer exists.
         """
-        if hasattr(self._session._conn, "table_registry"):
-            self._session._conn.table_registry.drop_table(self.table_name)
+        if hasattr(self._session._conn, "entity_registry"):
+            self._session._conn.entity_registry.drop_table(self.table_name)
```

Comment:
we only have the `entity_registry`/`table_registry` attribute on mock connection right?